lib/rubocop/cop/style/method_call_with_args_parentheses.rb

```diff
@@ -245,7 +245,8 @@ def call_in_literals?(node)
         def call_in_logical_operators?(node)
           node.parent &&
             (logical_operator?(node.parent) ||
-             node.parent.descendants.any?(&method(:logical_operator?)))
+             node.parent.send_type? &&
+             node.parent.arguments.any?(&method(:logical_operator?)))
```

Comment:
I think some other cop(s) had logic for complex conditions as well, so you might do some grepping and reuse/duplicate it to be on the safe side. I guess it should be `AssignmentInCondition` or something like this.

Reply:
The problem was around the use of logic operators in send arguments in the conditions and the bad check for it before this change. I haven't found any similar logic in `AssingmentInCondition` though. Maybe close enough for the other change that checks for assignment in chained calls? 

I'll see what I can dedup from this change. We have common utilities to the `MethodCallWithoutArgsParamethers` cop as well.